src/language/concurrency.md

```diff
@@ -237,303 +287,272 @@ After handling the events, the isolate exits.
-  Use `Isolate.run()` when targeting native platforms only,
-  for a more ergonomic API.
-{{site.alert.end}}
-
-[native and non-native platforms]: /overview#platform
-[Flutter's `compute` function]: {{site.flutter-api}}/flutter/foundation/compute.html
+(reading and writing files, for example), set timers, and more. It has its own
+memory and doesn't share any state with the main isolate. The worker isolate can
+block without affecting other isolates.
 
-#### Running an existing method in a new isolate
+### Isolate syntax
 
-The main isolate contains the code that spawns a new isolate: 
+There are two ways to work with isolates in Dart, depending on the use-case: 
 
-<?code-excerpt "lib/simple_worker_isolate.dart (main)"?>
-```dart
-const String filename = 'with_keys.json';
+* Use [`Isolate.run()`][] to perform a single computation on a separate thread.
+* Use [`Isolate.spawn()`][] to create an isolate that will handle
+  multiple messages over time, or a background worker.
 
-void main() async {
-  // Read some data.
-  final jsonData = await Isolate.run(_readAndParseJson);
+In most cases, `Isolate.run` is the recommended 
+API to run processes in the background.
 
-  // Use that data.
-  print('Number of JSON keys: ${jsonData.length}');
-}
-```
+### `Isolate.run()`
 
-The spawned isolate executes the function
-passed as the first argument, `_readAndParseJson`:
+The static `Isolate.run()` method requires one argument: a callback that will be 
+run on the newly spawned isolate.
 
-<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+<?code-excerpt "lib/isolate_run_syntax.dart"?>
 ```dart
-Future<Map<String, dynamic>> _readAndParseJson() async {
-  final fileData = await File(filename).readAsString();
-  final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
-  return jsonData;
+int slowFib(int n) => n <= 1 ? 1 : slowFib(n - 1) + slowFib(n - 2);
+
+// Compute without blocking current isolate.
+void fib40() async {
+  var result = await Isolate.run(() => slowFib(40));
+  print('Fib(40) = $result');
 }
 ```
 
-1. `Isolate.run()` spawns an isolate, the background worker,
-   while `main()` waits for the result.
-
-2. The spawned isolate executes the argument passed to `run()`:
-   the function `_readAndParseJson()`.
-
-3. `Isolate.run()` takes the result from `return`
-   and sends the value back to the main isolate,
-   shutting down the worker isolate.
-
-4. The worker isolate *transfers* the memory holding the result
-   to the main isolate. It *does not copy* the data.
-   The worker isolate performs a verification pass to ensure
-   the objects are allowed to be transferred.
-
-`_readAndParseJson()` is an existing,
-asynchronous function that could just as easily
-run directly in the main isolate.
-Using `Isolate.run()` to run it instead enables concurrency.
-The worker isolate completely abstracts the computations
-of `_readAndParseJson()`. It can complete without blocking the main isolate.
+#### Isolate.spawn
```

Comment:
Do you want this in code syntax to be consistent with the `Isolate.run` section?

```suggestion
#### `Isolate.spawn`
```